server/container/guice/blob-objectstorage-guice/pom.xml

```diff
@@ -35,6 +35,11 @@
     <description>An advanced email server - Object storage based Blob Store bindings</description>
 
     <dependencies>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-mailbox-deleted-messages-vault</artifactId>
```

Comment:
We should not have that dependency.

I understand that you need to be doing some cleanups and thus to know the buckets used by the vault in order to wipe them up.

However, this problem can easily be done with a higher level of indirection.

What I propose to you is to create a separated `DeletedMessageVaultCleaupTask` within tests cassandra-rabbit product. We don't need to have all of the cleanup done in one place right?

Another approach would be a feature independant cleanup, and having ObjectStorageDAO class being able to list buckets by reading swift metadata....

Reply:
okay! then I will implement this
> Another approach would be a feature independant cleanup, and having ObjectStorageDAO class being able to list buckets by reading swift metadata....